### PR TITLE
Doc/tut/azure: Add Azure tutorial

### PR DESCRIPTION
**Pull request by joshix**

### CODE REVIEW

File: Documentation/tutorials/azure/first-app.md [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+
+First, use `kubectl get` to view the deployment above as a manifest:
+
+```sh
+$ kubectl get deployment/simple-deployment -o yaml
+```
+
+The output YAML demonstrates that the `kubectl run` command created a resource in the cluster (which may be returned as YAML or JSON). These files can be used to create new resources, and may be stored in git, making them easily versionable.
+
+Before moving on, remove the deployment just created using `kubectl delete`:
```
Comment (athai) [inline Documentation/tutorials/azure/first-app.md]: Rephrase to "Remove the deployment *and service* just created..." since the following code example is removing both.

Should the Ingress resource also be deleted? Since the entire app is being redeployed in the next section? If yes, remember to update the following code snippet.

File: Documentation/tutorials/azure/check-logs.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Checking application logs
+
+Tectonic provides access to applications' logs both in Console and using `kubectl`.
```
Comment (coresolve) [inline Documentation/tutorials/azure/check-logs.md]: Applications aren't people. They can't own property :)   should be applications not applications'

File: Documentation/tutorials/azure/first-app.md [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+    <img src="img/azure-simple-ingress.png">
+    <div class="co-m-screenshot-caption">Viewing the Ingress in the Console</div>
+  </div>
+</div>
+
+Go to *Workloads > Deployments* and click on the Deployment name to monitor the deployment’s Pods.
+
+<div class="row">
+  <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 col-sm-12 col-xs-12 co-m-screenshot">
+    <img src="img/azure-simple-deployment.png">
```
Comment (athai) [inline Documentation/tutorials/azure/first-app.md]: nit: Image is of the deployment's overview. If we're telling the user that they can monitor pods through the console, the image should show the deployment "Pods" tab.

File: Documentation/tutorials/azure/check-logs.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    <div class="co-m-screenshot-caption">Logs for our simple Cookie Shop app</div>
+  </div>
+</div>
+
+Console will display available information for the pod, including CPU, memory, and logging stats. Click *Logs* to confirm the output is the same as that shown using `kubectl logs`.
+
+## Use kubectl to check logs
+
+Use `kubectl get pods` to view the cluster's pods.
+
```
Comment (coresolve) [inline Documentation/tutorials/azure/check-logs.md]: I wonder if we should be talking about namespaces here. This is all happening in the default namespace but we don't make that clear in our language.

File: Documentation/tutorials/azure/first-app.md (hunk 1 above)
Comment (joshix) [reply Documentation/tutorials/azure/first-app.md]: We want to leave the service and ingress.

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+* Copy and paste the contents of [`simple-service.yaml`][first-app-service-yaml] described in [Deploying an application from YAML][first-yaml] in to the editor, to replace the YAML file displayed.
+* Click *Create* to create the service, and open the *Service Overview* window.
+
+### Add Ingress
+
+Then, create ingress to the service:
+
+* Go to *Routing > Ingress* and click *Create Ingress*.
+* Copy and paste the contents of [`simple-ingress.yaml`][first-app-ingress-yaml] described in [Deploying an application from YAML][first-yaml] in to the editor, to replace the YAML file displayed.
+* Click *Create* to add the Ingress and open the Ingress *Overview* window.
```
Comment (athai) [inline Documentation/tutorials/azure/rolling-deployments.md]: Ingress -> Ingress resource

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 6]
```diff
@@ -0,0 +1,295 @@
...
+
+### Deploy v1.0
+
+When executing a rolling update, the rate at which updates will occur, and health checks to determine when an app has launched correctly may be defined.
```
Comment (athai) [inline Documentation/tutorials/azure/rolling-deployments.md]: "...rate at which updates will occur, and health checks..." <-- Remove the comma before "and"

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 7]
```diff
@@ -0,0 +1,295 @@
...
+
+First, edit the [`simple-deployment.yaml`][first-app-deployment-yaml] file used in [Deploying an application tutorial][first-app] to create a Service which more explicitly defines how Kubernetes will run the app.
```
Comment (athai) [inline Documentation/tutorials/azure/rolling-deployments.md]: "...to create a Service which more..." <-- Did you mean to type Deployment here? 

If no, please provide a sentence explaining why the user needs to edit `simple-deployment.yaml` to create a service.

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 8]
```diff
@@ -0,0 +1,295 @@
...
+
+To edit the Deployment:
+
+* Set the update model to `RollingUpdate`.
+* Add a `readiness` and `liveness` probe which instruct Kubernetes when and how to check that the Pods are alive and well – in this example, by checking for an HTTP response.
```
Comment (athai) [inline Documentation/tutorials/azure/rolling-deployments.md]: nit: I'd split the instruction (Add a `readiness` and `liveness` probe) and explanation (everything after that) up into two sentences. The current phrasing is somewhat awkward to read.

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 9]
```diff
@@ -0,0 +1,295 @@
...
+             port: 80
+             scheme: HTTP
+      dnsPolicy: ClusterFirst
+      restartPolicy: Always
+      terminationGracePeriodSeconds: 30
+```
+
+Click *Save Changes* to update the app and begin the rolling update.
+
+Click the *Pods* tab to see existing pods, and watch the live update in progress. Tectonic Console provides live updates as newer pods appear, and the corresponding older pods disappear.
```
Comment (athai) [inline Documentation/tutorials/azure/rolling-deployments.md]: "...live updates as newer pods appear, and the corresponding..." <-- Remove the comma before "and"

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 10]
```diff
@@ -0,0 +1,295 @@
...
+service "simple-service" created
+$ kubectl get services -o wide
+NAME              CLUSTER-IP   EXTERNAL-IP    PORT(S)        AGE    SELECTOR
+simple-service    10.3.113.190 <pending>      80:30657/TCP   1d     k8s-app=simple
+```
+
+Then, create the ingress on the cluster:
+```sh
+$ kubectl create -f simple-ingress.yaml
+ingress "simple-deployment" created
```
Comment (athai) [inline Documentation/tutorials/azure/rolling-deployments.md]: simple-deployment -> simple-ingress

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 11]
```diff
@@ -0,0 +1,295 @@
...
+$ kubectl get ingress
+NAME                HOSTS                               ADDRESS   PORTS     AGE
+simple-deployment   my-cluster.example.com             80        24s
```
Comment (athai) [inline Documentation/tutorials/azure/rolling-deployments.md]: simple-deployment -> simple-ingress

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 12]
```diff
@@ -0,0 +1,295 @@
...
+To update the app, first edit the `simple-deployment.yaml`file, as described above, to change
+  `image: quay.io/coreos/example-app:v1.0`
+to
+  `image: quay.io/coreos/example-app:v2.0`
+
+Then, open the deployment in Tectonic Console. The Console shows live updates, refreshing as listed pods are created and destroyed by the deployment. Click the *Pods* tab to see existing pods.
+
+<div class="row">
+  <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 col-sm-12 col-xs-12 co-m-screenshot">
+    <img src="img/azure-simple-service.png">
```
Comment (athai) [inline Documentation/tutorials/azure/rolling-deployments.md]: Wrong image is being used here. This is an image of the service, rather than deployment pods.

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 13]
```diff
@@ -0,0 +1,295 @@
...
+Tectonic Console provides live updates as newer pods appear, and the corresponding older pods disappear.
+
+<div class="row">
+  <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 col-sm-12 col-xs-12 co-m-screenshot">
+    <img src="img/azure-simple-deploy-updating.png">
+    <div class="co-m-screenshot-caption">Pods being created and terminated by Tectonic</div>
+  </div>
+</div>
+
+Reload the Cookie Shop page as the deployment updates to see changes.
```
Comment (athai) [inline Documentation/tutorials/azure/rolling-deployments.md]: Insert an image of Cookie Shop v2.0 after this sentence.

File: Documentation/tutorials/azure/check-logs.md [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+Copy one of the pod's names, and append it to the `kubectl logs -f ` command.
+
+```sh
+kubectl logs -f simple-deployment-4098151155-n8bqr
+10.2.1.1 - - [15/Aug/2017:21:30:32 +0000] "GET / HTTP/1.1" 200 576 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/60.0.3112.90 Safari/537.36" "127.0.0.1"
+```
+
+Open the app in your browser by visiting [my-cluster.example.com/simple-deployment][visit-app]. Note that there is a one in three chance that any visit to the page will hit any one of the three available pods. Refresh the page several times to increase the likelihood that the pod selected for kubectl logs will include a generated log.
+
+Use `kubectl logs -f` to review logs for the app. The `-f` flag tells kubectl to follow the logs. Exit this mode by bringing the terminal into focus and pressing CTRL-C.
```
Comment (athai) [inline Documentation/tutorials/azure/check-logs.md]: nit: While the explanation is useful, this first sentence is redundant (especially since the user has already used `kubectl logs -f` to review logs by this point). I'd rephrase this a bit.

File: Documentation/tutorials/azure/monitoring.md [hunk 15]
```diff
@@ -0,0 +1,210 @@
...
+
+```
+
+Finally, create `Ingress` to enable access to Prometheus from outside the cluster. Copy the following into a file named `prometheus-ingress.yaml`:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: prom-ingress
```
Comment (athai) [inline Documentation/tutorials/azure/monitoring.md]: Forgot to define a namespace here

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 16]
```diff
@@ -0,0 +1,295 @@
...
+
+To create a service using Tectonic Console:
+
+* Go to *Routing > Services* and click *Create Service*.
+* Copy and paste the contents of [`simple-service.yaml`][first-app-service-yaml] described in [Deploying an application from YAML][first-yaml] in to the editor, to replace the YAML file displayed.
+* Click *Create* to create the service, and open the *Service Overview* window.
+
+### Add Ingress
+
+Then, create ingress to the service:
```
[the remark this answers is not in the document]
Comment (joshix) [reply Documentation/tutorials/azure/rolling-deployments.md]: yes and in the edit I think we have enough context to just say "Ingress" w/o the "resource"

File: Documentation/tutorials/azure/deleting-deployment.md [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+
+```sh
+$ kubectl delete deploy/simple-deployment svc/simple-service
+deployment "simple-deployment" deleted
+service "simple-service" deleted
+```
+
+This will delete the Deployment (including Pods and Replica Sets) and the Service associated with the `simple` app created in the [Deploying an application on Tectonic][first-app] tutorial.
+
+As of version 1.7, Ingress resources no longer appear in the `all` group. Use the same sequence of `kubectl get ing` and `kubectl delete` to delete the Ingress resource.
```
Comment (athai) [inline Documentation/tutorials/azure/deleting-deployment.md]: `kubectl get ing` -> `kubectl get ingress`

File: Documentation/tutorials/azure/index.md [hunk 18]
```diff
@@ -0,0 +1,20 @@
...
+CoreOS Tectonic is enterprise-ready Kubernetes. These step-by-step tutorials show how to get started with Tectonic on Azure. They describe:
+
+* [Installing Tectonic on Microsoft Azure][install-azure]
+* [Deploying an application on the Tectonic cluster][first-app]
+* [Checking application logs][check-logs]
+* [Scaling an application][scale-app]
+* [Versioning and updating an application][rolling-deployments]
+* [Monitoring an application with Prometheus in Tectonic Console][monitoring-apps]
+
+<a href="install.md" class="btn btn-primary btn-lg">Let's Get Started</a>
```
Comment (athai) [inline Documentation/tutorials/azure/index.md]: This doesn't actually work. See https://coreos.com/tectonic/docs/latest/tutorials/azure

File: Documentation/tutorials/azure/install.md [hunk 19]
```diff
@@ -0,0 +1,223 @@
...
+
+## Configuring Azure DNS
+
+The Azure DNS service allows you to perform DNS management, traffic management, availability monitoring and domain registration. DNS management is the only feature of Azure DNS required to install Tectonic.
+
+### Create an Azure DNS Zone
+
+When creating an Azure DNS Zone, enter a domain or subdomain that you own and can manage.
+
+The Tectonic installation requires a domain or subdomain name in which it will create two [sub-]subdomains; one for the Tectonic console, and one for the Kubernetes API server. This allows Tectonic to access and use the listed domain. This tutorial employs the domain name `example.com`. The string `example.com` should be replaced with the domain or subdomain name configured in this step wherever it appears later in the tutorial.
```
Comment (athai) [inline Documentation/tutorials/azure/install.md]: "...it will create two [sub-]subdomains; one for the Tectonic..." <-- Change the semicolon to a colon

File: Documentation/tutorials/azure/install.md [hunk 20]
```diff
@@ -0,0 +1,223 @@
...
+
+1. From the menu at left, selet *DNS Zones*
+2. Click *Add* to create a new zone.
+3. Enter an existing, registered domain or subdomain name.
+4. Click *Create*.
+
+Azure provides 4 DNS nameservers for the new zone. The domain or sub-domain must be [configured to use these nameservers][azure-dns-delegate]. Visit the domain registrar to add the Route53 NS records.
```
Comment (athai) [inline Documentation/tutorials/azure/install.md]: Remove the reference to Route53.

File: Documentation/tutorials/azure/rolling-deployments.md (hunk 13 above)
Comment (joshix) [reply Documentation/tutorials/azure/rolling-deployments.md]: with it used above in console example, foregoing here

File: Documentation/tutorials/azure/install.md [hunk 21]
```diff
@@ -0,0 +1,223 @@
...
+{
+ "appId": "generated-app-id",
+ "displayName": "azure-cli-2017-01-01",
+ "name": "http://tectonic-coreos",
+ "password": "generated-pass",
+ "tenant": "generated-tenant"
+}
+```
+
+Export the following environment variables with values obtained from the output of the SP role assignment. As noted above, `ARM_SUBSCRIPTION_ID` is the `id` of the Azure account returned by `az login`.
```
Comment (athai) [inline Documentation/tutorials/azure/install.md]: I don't think `ARM_SUBSCTIPTION_ID` is actually noted above?

File: Documentation/tutorials/azure/install.md [hunk 22]
```diff
@@ -0,0 +1,223 @@
...
+```
+
+Edit the parameters in `build/$CLUSTER/terraform.tfvars` with the deployment's Azure details, domain name, license, and pull secret. [View all of the Azure specific options][azure-vars] and [the common Tectonic variables][vars].
+
+### Key values for basic Azure deployment
+
+These are the basic values that must be adjusted for each Tectonic deployment on Azure. See the details of each value in the comments in the `terraform.tfvars` file.
+
+* `tectonic_admin_email` - For the initial Console login
+* `tectonic_admin_password_hash` - Bcrypted value
```
Comment (coresolve) [inline Documentation/tutorials/azure/install.md]: Maybe a little more about "Bcrypted value"

File: Documentation/tutorials/azure/scale-app.md [hunk 23]
```diff
@@ -0,0 +1,142 @@
...
+  replicas: 5   # bump from 3 to 5
+```
+
+Click *Save Changes* to scale the deployment to the new number of pods.
+
+## Monitoring deployment scale in Tectonic Console
+
+Use the *Pods* tab of a given deployment to track the number of replicas, and their health.
+
+## Scaling a deployment with kubectl
```
[the remark this answers is not in the document]
Comment (joshix) [reply Documentation/tutorials/azure/scale-app.md]: //TODO

File: Documentation/tutorials/azure/monitoring.md [hunk 24]
```diff
@@ -0,0 +1,210 @@
...
+Next, create a means to access the Prometheus UI. Copy the following into a file named `prometheus-service.yaml`:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  name: prometheus-hello-world
+  namespace: default
+spec:
+  type: LoadBalancer
```
Comment (coresolve) [inline Documentation/tutorials/azure/monitoring.md]: If it's going to be behind an ingress. This should not be LoadBalancer it should be NodePort. 

The key diff is that type:LoadBalancer will create an public lb and expose the service directly. You shouldn't do this unless you are going to use that as a way to access the app. Otherwise your just throwing money down the tube and exposing your app in a bad way.

File: Documentation/tutorials/azure/rolling-deployments.md (hunk 13 above)
Comment (athai) [reply Documentation/tutorials/azure/rolling-deployments.md]: Yes, but why show the image for v1 twice but not v2?

File: Documentation/tutorials/azure/index.md (hunk 18 above)
Comment (joshix) [reply Documentation/tutorials/azure/index.md]: trailing slash
see https://coreos.com/tectonic/docs/latest/tutorials/azure/

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 25]
```diff
@@ -0,0 +1,295 @@
+# Versioning and updating an application on Tectonic
+
+Kubernetes uses rolling deployments to update an application between releases. When a deployment configuration is updated, Tectonic automatically starts to execute a smooth, rolling update of the application. If the update fails it can roll back to the previous instance.
```
Comment (coresolve) [inline Documentation/tutorials/azure/rolling-deployments.md]: Kubernetes Deployment Objects by default use rollingupdate as the way to make changes to pods.

File: Documentation/tutorials/azure/install.md (hunk 21 above)
Comment (joshix) [reply Documentation/tutorials/azure/install.md]: line 60? or too unclear/distant?

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 26]
```diff
@@ -0,0 +1,295 @@
...
+* Deploy a sample "Cookie Shop" application v1.0.
+* Update the sample application from v1.0 to v2.0 without app downtime.
+* Perform a health check on the sample application.
+* Roll back to v1.0.
+
+## Updating an app using Tectonic Console
+
+Use Tectonic Console to update an active application and demonstrate the power of rolling updates.
+
+### Create a Service
```
Comment (coresolve) [inline Documentation/tutorials/azure/rolling-deployments.md]: use `kubectl explain service`

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 27]
```diff
@@ -0,0 +1,295 @@
...
+
+If the service created using `simple-service.yaml` and `simple-ingress.yaml` from the [Deploying an application tutorial][first-app] is running, it may be reused for this tutorial, and its deployment updated. If not, create the service before continuing with this tutorial.
+
+To create a service using Tectonic Console:
+
+* Go to *Routing > Services* and click *Create Service*.
+* Copy and paste the contents of [`simple-service.yaml`][first-app-service-yaml] described in [Deploying an application from YAML][first-yaml] in to the editor, to replace the YAML file displayed.
+* Click *Create* to create the service, and open the *Service Overview* window.
+
+### Add Ingress
```
Comment (coresolve) [inline Documentation/tutorials/azure/rolling-deployments.md]: use `kubectl explain ingress`

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 28]
```diff
@@ -0,0 +1,295 @@
...
+
+Then, create ingress to the service:
+
+* Go to *Routing > Ingress* and click *Create Ingress*.
+* Copy and paste the contents of [`simple-ingress.yaml`][first-app-ingress-yaml] described in [Deploying an application from YAML][first-yaml] in to the editor, to replace the YAML file displayed.
+* Click *Create* to add the Ingress and open the Ingress *Overview* window.
+
+### Deploy v1.0
```
Comment (coresolve) [inline Documentation/tutorials/azure/rolling-deployments.md]: use `kubectl explain deployment.spec.template`

example:
```
$ kubectl explain deployment.spec.strategy.type
FIELD: type <string>

DESCRIPTION:
     Type of deployment. Can be "Recreate" or "RollingUpdate". Default is
     RollingUpdate.
git: enable_psp
dcooley@lynx ~/go/src/github.com/coreos/tectonic-installer
$ kubectl explain deployment.spec.strategy.rollingUpdate
RESOURCE: rollingUpdate <Object>

DESCRIPTION:
     Rolling update config params. Present only if DeploymentStrategyType =
     RollingUpdate.

    Spec to control the desired behavior of rolling update.

FIELDS:
   maxSurge	<string>
     The maximum number of pods that can be scheduled above the desired number
     of pods. Value can be an absolute number (ex: 5) or a percentage of desired
     pods (ex: 10%). This can not be 0 if MaxUnavailable is 0. Absolute number is
     calculated from percentage by rounding up. By default, a value of 1 is used.
     Example: when this is set to 30%, the new RC can be scaled up immediately
     when the rolling update starts, such that the total number of old and new
     pods do not exceed 130% of desired pods. Once old pods have been killed, new
     RC can be scaled up further, ensuring that total number of pods running at
     any time during the update is atmost 130% of desired pods.

   maxUnavailable	<string>
     The maximum number of pods that can be unavailable during the update. Value
     can be an absolute number (ex: 5) or a percentage of desired pods (ex: 10%).
     Absolute number is calculated from percentage by rounding down. This can not
     be 0 if MaxSurge is 0. By default, a fixed value of 1 is used. Example: when
     this is set to 30%, the old RC can be scaled down to 70% of desired pods
     immediately when the rolling update starts. Once new pods are ready, old RC
     can be scaled down further, followed by scaling up the new RC, ensuring that
     the total number of pods available at all times during the update is at
     least 70% of desired pods.
```

File: Documentation/tutorials/azure/rolling-deployments.md (hunk 13 above)
Comment (joshix) [reply Documentation/tutorials/azure/rolling-deployments.md]: true enough...

File: Documentation/tutorials/azure/rolling-deployments.md [hunk 29]
```diff
@@ -0,0 +1,295 @@
...
+
+Reload the Cookie Shop page as the deployment updates to see the changes:
+
+<div class="row">
+  <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1 col-sm-12 col-xs-12">
+    <img src="img/azure-cookie-v2.png">
+  </div>
+</div>
+
+### Perform a health check
```
Comment (coresolve) [inline Documentation/tutorials/azure/rolling-deployments.md]: use
```
kubectl explain deployment.spec.template.spec.containers.livenessProbe
kubectl explain deployment.spec.template.spec.containers.readinessProbe
```

File: Documentation/tutorials/azure/scale-app.md [hunk 30]
```diff
@@ -0,0 +1,142 @@
+# Scaling an application on Tectonic
+
+Kubernetes can scale applications either statically, based on an explicit count of replicas, or dynamically, adding and removing replicas based on application load.
```
Comment (coresolve) [inline Documentation/tutorials/azure/scale-app.md]: call out to docs here as a reader I want a link where I can go read more about "based on application load model"

File: Documentation/tutorials/azure/install.md (hunk 21 above)
Comment (athai) [reply Documentation/tutorials/azure/install.md]: Ah, okay. It should be good then. Didn't actually type in `az login`, but I guess that's where the output is coming from?

File: Documentation/tutorials/azure/first-app.md [hunk 31]
```diff
@@ -0,0 +1,298 @@
...
+  ports:
+  - protocol: TCP
+    port: 80
+  type: NodePort
+```
+
+To connect the service to the containers run by the deployment, the deployment `containerPort` and the service `port` must match.
+
+Copy the following into a file named `simple-ingress.yaml`. This file will be used to create an Ingress resource to act as a local load balancer.
+
```
[the remark this answers is not in the document]
Comment (joshix) [reply Documentation/tutorials/azure/first-app.md]: I believe we did that to avoid a trip back to DNS configuration.

File: Documentation/tutorials/azure/first-app.md [hunk 32]
```diff
@@ -0,0 +1,298 @@
...
+2. Click *Mac* or *Linux* to download the `kubectl` binary for your operating system.
+3. Click *I’m Done* to exit the download window and return to the console.
+
+Move the downloaded `kubectl` file to `/usr/local/bin` or another directory on the shell's PATH:
+
+```sh
+$ chmod +x kubectl
+$ mv kubectl /usr/local/bin/kubectl
+```
+
```
[the remark this answers is not in the document]
Comment (joshix) [reply Documentation/tutorials/azure/first-app.md]: I (somewhat arbitrarily) chose one of the ways I knew to set kubeconfig. I didn't want an intro tutorial freighted with more than one way to do it (with apologies to Larry Wall).

File: Documentation/tutorials/azure/install.md (hunk 21 above)
Comment (joshix) [reply Documentation/tutorials/azure/install.md]: <resolved>

File: Documentation/tutorials/azure/first-app.md [hunk 33]
```diff
@@ -0,0 +1,298 @@
...
+### Acquiring Tectonic configuration and credentials for kubectl
+
+Begin by logging in to Tectonic Console to download a `kubeconfig` file and the `kubectl` program:
+
+1. Log in to Tectonic Console at `https://my-cluster.example.com`
+1. From Tectonic Console, click *Tectonic Admin > My Account* on the bottom left of the page.
+2. Click KUBECTL: *Download Configuration*, and follow the onscreen instructions to authenticate.
+3. When the *Set Up kubectl* window opens, click *Verify Identity*.
+4. Enter the username and password and click *Login*.
+5. Copy the alphanumeric string on the *Login Successful* screen.
```
[the remark this answers is not in the document]
Comment (joshix) [reply Documentation/tutorials/azure/first-app.md]: noted in sentence introducing the numbered list

File: Documentation/tutorials/azure/first-app.md (hunk 1 above)
Comment (athai) [reply Documentation/tutorials/azure/first-app.md]: (resolved)

**Comment by coresolve** [comment]
LGTM
